### PR TITLE
Update modal docs: add built-in response section

### LINKED ISSUE #2127: Provide redirect feature for modal buttons
### Describe the feature in detail (code, mocks, or screenshots encouraged)

In many situations, buttons such as 'cancel', 'confirm', and 'submit' require redirection to a different page. Introducing default variables like `buttonHrefCancel`, `buttonHrefConfirm`, and `buttonHrefSubmit` in the standard types (alert/confirm/prompt) would be advantageous. This concept is akin to text override options found here: https://github.com/skeletonlabs/skeleton/blob/dev/packages/skeleton/src/lib/utilities/Modal/types.ts#L38.

### What type of pull request would this be?

None

### Provide relevant links or additional information.

_No response_

### PR DESCRIPTION
**Pull request by phamduylong**
## Linked Issue

Closes #2127

## Description

Add built-in response section to docs

## Changsets

Instructions: Changesets automate our changelog. If you modify files in `/packages`, run `pnpm changeset` in the root of the monorepo, follow the prompts, then commit the markdown file. Changes that add features should be `minor` while chores and bugfixes should be `patch`. Please prefix the changeset message with `feat:`, `bugfix:` or `chore:`.

## Checklist

Please read and apply all [contribution requirements](https://www.skeleton.dev/docs/contributing).

- [x] This PR targets the `dev` branch (NEVER `master`)
- [x] Documentation reflects all relevant changes
- [x] Branch is prefixed with: `docs/`, `feat/`, `chore/`, `bugfix/`
- [x] Ensure Svelte and Typescript linting is current - run `pnpm ci:check`
- [x] Ensure Prettier linting is current - run `pnpm format`
- [x] All test cases are passing - run `pnpm test`
- [ ] Includes a changeset (if relevant; see above)

### CODE REVIEW

**Comment by endigo9740** [comment]
@phamduylong I think you're on the right track, but I think we might need to adjust this a bit. I'm short on time today, but I'll write up a proper explanation soon. Thanks for your patience.

File: sites/skeleton.dev/src/routes/(inner)/utilities/modals/+page.svelte [hunk 1]
```diff
@@ -598,6 +598,77 @@ const modalComponent: ModalComponent = {
...
+						<p>For alert modals, no response will be returned. The modal will simply close when the user clicks the "Cancel" button.</p>
+					</svelte:fragment>
+				</AccordionItem>
+				<AccordionItem>
+					<svelte:fragment slot="summary"><h3 class="h3" data-toc-ignore>Confirm Modal</h3></svelte:fragment>
+					<svelte:fragment slot="content">
+						<p>
+							Confirm modals, as the name implies, will return either <code class="code">true</code> or <code class="code">false</code>.
+							Clicking the "OK" button will return <code class="code">true</code>, while other interactions such as pressing "Cancel" or
+							using the backdrop will return <code class="code">false</code>.
```
Comment (Mahmoud-zino) [inline sites/skeleton.dev/src/routes/(inner)/utilities/modals/+page.svelte:618]: using the backdrop will return undefined!

Comment (phamduylong) [reply sites/skeleton.dev/src/routes/(inner)/utilities/modals/+page.svelte]: Great catch, thanks. I wasn't cautious with what I was writing.

**Comment by endigo9740** [comment]
@phamduylong I've gone in a slightly different direction with this. This info is now part of the main accordion and is explained just before handling your own custom response value.

Screenshot 2023-12-18 at 3 00 13 PM

**Pull request merged**